Predictive gfm 2025 fix examples

allaffa opened the pull request.

RylieWeaver reviewed (commented):
All changes related to the following look good to me:
(1) Periodic Boundary Conditions: including RadiusGraphPBC, PBC transforms (e.g. PBCDistance), and examples preprocessing.
(2) compute_grad_energy and its node output dim in `hydragnn/utils/input_config_parsing/config_utils.py` and `examples/multibranch/train.py`

This PR accurately and completely reflects the changes I made or recommended for those two components.

RylieWeaver commented:
@allaffa 

I think we need to add the Fairchem package to an install file because the checks show the following error. Do you know the exact version from your environment that was working?:

```
==================================== ERRORS ====================================
_____ ERROR collecting examples/omat24/fairchem/core/common/test_utils.py ______
ImportError while importing test module '/home/runner/work/HydraGNN/HydraGNN/examples/omat24/fairchem/core/common/test_utils.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.9.22/x64/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
examples/omat24/fairchem/core/__init__.py:12: in <module>
    from fairchem.core.common.relaxation.ase_utils import OCPCalculator
E   ModuleNotFoundError: No module named 'fairchem'
=========================== short test summary info ============================
ERROR examples/omat24/fairchem/core/common/test_utils.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 7.91s ===============================
```

allaffa commented:
> @allaffa
> 
> I think we need to add the Fairchem package to an install file because the checks show the following error. Do you know the exact version from your environment that was working?:
> 
> ```
> ==================================== ERRORS ====================================
> _____ ERROR collecting examples/omat24/fairchem/core/common/test_utils.py ______
> ImportError while importing test module '/home/runner/work/HydraGNN/HydraGNN/examples/omat24/fairchem/core/common/test_utils.py'.
> Hint: make sure your test modules/packages have valid Python names.
> Traceback:
> /opt/hostedtoolcache/Python/3.9.22/x64/lib/python3.9/importlib/__init__.py:127: in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
> examples/omat24/fairchem/core/__init__.py:12: in <module>
>     from fairchem.core.common.relaxation.ase_utils import OCPCalculator
> E   ModuleNotFoundError: No module named 'fairchem'
> =========================== short test summary info ============================
> ERROR examples/omat24/fairchem/core/common/test_utils.py
> !!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
> =============================== 1 error in 7.91s ===============================
> ```

@RylieWeaver 
I removed the file. The issue with the file is that `pytest` identifies all python scripts that contain the word `test` in their name, and automatically  interpret it as a python script with tests to run during the CI testing. Obviously, these tests are not things that we want to run ourselves. Therefore, I removed the file.

allaffa commented:
@kshitij-v-mehta @jychoi-hpc @pzhanggit 

This PR authored by me and @RylieWeaver  mainly changes/adds new examples. The edits to existing examples mainly involve fixing how periodic boundary conditions are implemented and applied to the PyG data objects. 

In the `examples/omat24` I copied a substantial part of the utilities of the repository [fairchem](https://github.com/facebookresearch/fairchem). I would appreciate any advice you could give me in acknowledging the proper credit. On the one hand, I want to make sure that I use frozen versions of the data preprocessing routines (thereby my copy-paste) but I also want to make sure that I properly acknowledge the original provenance of the code.

RylieWeaver commented:
@kshitij-v-mehta @jychoi-hpc @pzhanggit

Additionally, please check that all components relating to branching and DDP are correct. The main directories I see for those subjects are `multibranch`, `multidataset`, and `multi dataset_deepspeed`.

pzhanggit commented:
> @kshitij-v-mehta @jychoi-hpc @pzhanggit
> 
> This PR authored by me and @RylieWeaver mainly changes/adds new examples. The edits to existing examples mainly involve fixing how periodic boundary conditions are implemented and applied to the PyG data objects.
> 
> In the `examples/omat24` I copied a substantial part of the utilities of the repository [fairchem](https://github.com/facebookresearch/fairchem). I would appreciate any advice you could give me in acknowledging the proper credit. On the one hand, I want to make sure that I use frozen versions of the data preprocessing routines (thereby my copy-paste) but I also want to make sure that I properly acknowledge the original provenance of the code.

Regarding `fairchem`, did you need to make changes to their implementation? If not, could we just use it as an installed package?

allaffa commented:
> > @kshitij-v-mehta @jychoi-hpc @pzhanggit
> > This PR authored by me and @RylieWeaver mainly changes/adds new examples. The edits to existing examples mainly involve fixing how periodic boundary conditions are implemented and applied to the PyG data objects.
> > In the `examples/omat24` I copied a substantial part of the utilities of the repository [fairchem](https://github.com/facebookresearch/fairchem). I would appreciate any advice you could give me in acknowledging the proper credit. On the one hand, I want to make sure that I use frozen versions of the data preprocessing routines (thereby my copy-paste) but I also want to make sure that I properly acknowledge the original provenance of the code.
> 
> Regarding `fairchem`, did you need to make changes to their implementation? If not, could we just use it as an installed package?

@pzhanggit 
In principle, you are 100% correct. We could install the package as a standalone module. However, I already experienced in the past that the developer of this package do not necessarily ensure backward compatibility with older versions. In some cases, they change the format in which their functions process the data. When this happens, I do not necessarily want this to affect our work.

RylieWeaver commented on examples/alexandria/train.py:
```diff
@@ -92,39 +99,36 @@ def __init__(
 
-        self.radius_graph = RadiusGraph(5.0, loop=False, max_num_neighbors=50)
-        self.radius_graph_pbc = RadiusGraphPBC(5.0, loop=False, max_num_neighbors=50)
+        self.radius_graph = RadiusGraph(10.0, loop=False, max_num_neighbors=10)
```
@allaffa 

Do you agree? If yes I will make this change in all the examples.

allaffa replied:
@RylieWeaver 
Yes, I agree

kshitij-v-mehta commented:
> @kshitij-v-mehta @jychoi-hpc @pzhanggit
> 
> This PR authored by me and @RylieWeaver mainly changes/adds new examples. The edits to existing examples mainly involve fixing how periodic boundary conditions are implemented and applied to the PyG data objects.
> 
> In the `examples/omat24` I copied a substantial part of the utilities of the repository [fairchem](https://github.com/facebookresearch/fairchem). I would appreciate any advice you could give me in acknowledging the proper credit. On the one hand, I want to make sure that I use frozen versions of the data preprocessing routines (thereby my copy-paste) but I also want to make sure that I properly acknowledge the original provenance of the code.

You have credited the original authors correctly as required by the MIT license of their code. I would recommend adding the release version or the Git commit ID of the fairchem code to track updates to their code.

allaffa commented on examples/qm7x/requirements.txt:
```diff
@@ -0,0 +1,2 @@
+# to uncompress .7z file after download
```
Since it is needed only for this example, wouldn't it be clearer to keep it in the example folder? 
@kshitij-v-mehta @jychoi-hpc  any thoughts?

kshitij-v-mehta replied:
I recommend putting it in requirements-optional, with a comment that it is only needed for the qm7x example.

allaffa replied:
Done. Thanks.

allaffa commented:
> > @kshitij-v-mehta @jychoi-hpc @pzhanggit
> > This PR authored by me and @RylieWeaver mainly changes/adds new examples. The edits to existing examples mainly involve fixing how periodic boundary conditions are implemented and applied to the PyG data objects.
> > In the `examples/omat24` I copied a substantial part of the utilities of the repository [fairchem](https://github.com/facebookresearch/fairchem). I would appreciate any advice you could give me in acknowledging the proper credit. On the one hand, I want to make sure that I use frozen versions of the data preprocessing routines (thereby my copy-paste) but I also want to make sure that I properly acknowledge the original provenance of the code.
> 
> You have credited the original authors correctly as required by the MIT license of their code. I would recommend adding the release version or the Git commit ID of the fairchem code in your README. That will make it easy to track updates to their code.

@kshitij-v-mehta 
Done. Thanks

allaffa commented on examples/alexandria/alexandria_energy.json:
```diff
@@ -4,10 +4,15 @@
   },
   "NeuralNetwork": {
     "Architecture": {
+      "global_attn_engine": "",
+      "global_attn_type": "",
```
@ArCho48 
Do you confirm that the code can handle situations where `global_attn_engine` and `global_attn_type` are missing from the JSON file?

ArCho48 replied:
In one word, yes. 
The code expects to see either GPS or "" for global_attn_engine. If global_attn_engine is "", it doesnt matter if global_attn_type is "" or has some value, it only matters when global_attn_engine is GPS.

ArCho48 replied:
No, create model config takes these two fields as mandatory arguments. Would you like me to make the change in config_utils to set their default to None.

allaffa replied:
@ArCho48 
Yes, please make these changes ASAP. Thanks.
